include/seqan3/range/container/aligned_sequence_adaptor_constant_access.hpp

```diff
@@ -0,0 +1,688 @@
+    * sequence representation.
+    *
+    * We use the rank support structure indexed for counting 1s and substract
+    * the number gaps in [0; position_gap] from the input position.
+    * E.g.           aligned sequence  | - A - - T
+    *                 position_gapped  | 0 1 2 3 4
+    *      map_to_underlying_position  |-1 0 0 0 1
+    * Note that a gap is mapped to the same position than the next preceeding
+    * non-gap symbol.
+    */
+    difference_type map_to_underlying_position(size_type const position_gapped) const
+    {
+        return static_cast<difference_type>(position_gapped) -
+        static_cast<difference_type>(rs.rank(std::min<size_type>(position_gapped+1, this->size())));
+    }
+    //!\}
+
+    /*!\name Random access sequence concept support.
+     * \{
+    */
+    //!\brief Return reference to aligned sequence for given index.
+    constexpr reference operator[](size_type const n) const // const noexcept(noexcept((*host)[pos+n]))
+    {
+        assert(n < size());
+        if (!gap_vector[n]){
+            size_type const pos = map_to_underlying_position(n);
+            return value_type(sequence[pos]);
+        }
+        return gap_symbol;
+    }
+
+    //!\brief Return reference to aligned sequence for given index.
+    value_type at(size_type const idx) const
+    {
+        return (value_type)(*this)[idx];
+    }
+
+    /*!\brief Resize the aligned sequence. Extend with gaps if the new sizes
+    * enlarges the current aligned sequence.
+    *
+    * Note that sdsl does not initialize int_vectors with 0s when applying resize.
+    */
+    void resize(size_type size)
+    {
+        size_type size_old = this->size();
+        gap_vector.resize(size);
+        for (size_type i = size_old; i < this->size(); ++i)
+            gap_vector[i] = 1;
+        // decrease compressed sequence eventually
+        if (size_old > 0)
+            sequence.resize(ss.select(size_old-1));
+        update_support_structures();
+    }
+    //{ val.resize(0, typename type::value_type{}) } -> void;
+    /*!\brief Resize the aligned sequence. Extend with gaps if the new sizes
+    * enlarges the current aligned sequence.
+    *
+    * Note that sdsl does not initialize int_vectors with 0s when applying resize.
+    */
+    void resize(size_type size, value_type value)
+    {
+        if (value == gap::GAP || size < this->size())
+            return resize(size);
+        // extend with non-gap value size()-size times
+        size_type size_old = this->size();
+        gap_vector.resize(size);
+        for (size_type i = size_old; i < this->size(); ++i)
+            gap_vector[i] = 0;
+        sequence.insert(sequence.cend(), this->size()-size_old, value);
+        update_support_structures();
+    }
+
+    //!\}
+private:
+    //!\privatesection
+    //!\brief Where the gap symbol is stored.
+    //!\hideinitializer
+    constexpr value_type static const gap_symbol = value_type(gap::GAP);
```

Comment:
Since you are actually returning copies and not references you can get rid of this, or not?